prog/collide.go

```diff
@@ -97,3 +97,40 @@ func DoubleExecCollide(origProg *Prog, rand *rand.Rand) (*Prog, error) {
 	prog.Calls = append(prog.Calls, dupCalls...)
 	return prog, nil
 }
+
+// Duplicate every call in the program and execute the duplicated copies in parallel.
+// To ensure that async call execute in order, set async to every other call.
+// Example input:
+// r1 = test1()
+// test2(r1)
+// test3(r1)
+// Example output:
+// r1 = test1() (async)
+// r2 = test1()
+// test2(r1) # Note that we don't assign async here.
```

Comment:
Why? I don't see anything in the code that would result in not assigning async for this call.

Reply:
`assignAsync` alternates every original call (i.e. it changes value every 2 calls in the resulting program).